aws/handlers/appointments_api_req_handler.go

```diff
@@ -17,15 +17,20 @@ func HandleAppointmentsRequest(req events.APIGatewayProxyRequest) (*events.APIGa
 	switch req.HTTPMethod {
 	case "GET":
 		db, err := database.CreateDB()
-
 		if err != nil {
 			return utils.GatewayResponse(http.StatusBadRequest, models.ErrorBody{ErrorMsg: aws.String(err.Error())})
 		}
 
 		repository := repositories.AppointmentsRepository{DB: db}
 		//TO-DO this is work in progress need to add api to get appointments per user/counselor/timeslot
-		appointments, err := repository.FetchAll(nil)
+		counselorId  := req.QueryStringParameters["counselor"]
```

Comment:
```suggestion
		counselorId  := req.QueryStringParameters["counselor"]
		if counselorId == nil {
		    // return error
		}
		
		counselorRepository := repositories.CounselorsRepository{DB: db}
		counselor, error := counselorRepository.Fetch(counselor)
		
		if err != nil {
		    // return no valid counselor found error
		}
		
		// Do the same for user, either by parsing user Id from the passed in cognito token
		
```

There is not much overhead from getting/validating user/counselor at least for now, cuz the database connection is established only once... so it should pretty fast to query data from different tables... regarding the other question, it will not be that straightforward to query for appointments by counselor/user Id if you think about it.. there will be many predicates that we would prefer to have .. like account freeze etc .. passing ids will not scale for such future use cases...

Reply:
i think having separate tables defeats the purpose of having nosql dynamodb . we need to store all data in single table with different primary secondary key there is no need to validate on a user id or counselor id like we do in relational database , https://www.trek10.com/blog/dynamodb-single-table-relational-modeling/  , letme know if you are ok with creating a single table